src/Datasource/EntityTrait.php

```diff
@@ -360,6 +360,20 @@ public function unsetProperty($property)
     }
 
     /**
+     * Get the list of all properties.
+     *
+     * The list of all properties is all standard properties
+     * plus virtual properties plus hidden properties.
+     *
+     * @return array A list of all entity properties
+     */
+    public function allProperties()
```

Comment:
@ndm2 You are right. Making `allProperties()` behave differently than other `*Properties()` would be confusing, so better to make it just returning property names. 

Though using `$entity->extract($entity->allProperties)` won't return results similar to `$entity->toArray()` as the former won't convert nested entities to array. So we would still need another method for that.